### PR TITLE
Docker container that builds nix scripts and integration tests

### PR DESCRIPTION
**Pull request by wiredhikari**
Bringing the power of nix to non-nix users

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -9,11 +9,16 @@ in naersk.buildPackage {
...
-  gitSubmodules = true;
+  copyTarget = true;
   buildInputs = [
       pkgs.openssl
       pkgs.pkg-config
       pkgs.perl
   ];
+  shellHook =
+  ''
+    echo "Hello shell"
```
Comment (justinmoon) [inline default.nix:20]: Is this necessary?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM nixos/nix
+ADD . .
+RUN nix-channel --update
+RUN nix-shell --command ./scripts/integrationtest.sh
```
Comment (justinmoon) [inline Dockerfile:4]: This dockerfile doesn't have a `CMD` or `ENTRYPOINT` so I don't think you can really run this multiple times. When you build the image with `docker build` it will run the integration tests *at build time*. When you run the container with `docker run` then nothing happens since there's no `CMD` or `ENTRYPOINT` ...

File: shell.nix [hunk 3]
```diff
@@ -22,6 +22,7 @@ pkgs.mkShell {
     bitcoin
     clightning
     jq
+    procps
```
Comment (justinmoon) [inline shell.nix:25]: Why was this added?

**Comment by justinmoon** [comment]
Also, the container is 14.2GB which seems kinda big but not super experienced with Docker / NixOS ...

File: default.nix (hunk 1 above)
Comment (wiredhikari) [reply default.nix:20]: Kept it so it will be easy to track when debugging github actions, but yeah we can remove it

File: shell.nix (hunk 3 above)
Comment (wiredhikari) [reply shell.nix:25]: `pkill` isn't being identified during docker build so added it as a package in nix-shell

**Comment by elsirion** [comment]
Probably needs a rebase for CI to run

**Review by justinmoon — APPROVED** [review]
LGTM. I think we can try to optimize the caching in future PRs. But this is a good step forward.

**Comment by dpc** [comment]
I would expect the main `Dockerfile` to give me a production build of the project to run in my infra or something. I would name this one `Dockerfile.integrationtest` or something to be ran with `docker -f Dockerfile.integrationtest .`.

**Comment by elsirion** [comment]
Good point @dpc
